### PR TITLE
Added jenkins support for windows 2022

### PR DESCRIPTION
**Pull request by salsal97**

### CODE REVIEW

File: .jenkins/infrastructure/e2e_testing.Jenkinsfile [hunk 1]
```diff
@@ -100,8 +103,11 @@ pipeline {
                     string(name: 'UBUNTU_2004_ICX_CUSTOM_LABEL', value: params.UBUNTU_2004_ICX_LABEL),
                     string(name: 'UBUNTU_2004_NONSGX_LABEL', value: params.UBUNTU_NONSGX_LABEL),
                     string(name: 'WS2019_DCAP_CFL_LABEL', value: params.WINDOWS_2019_DCAP_LABEL),
-                    string(name: 'WS2019_DCAP_ICX_LABEL', value: params.WINDOWS_2019_DCAP_LABEL),
+                    string(name: 'WS2019_DCAP_ICX_LABEL', value: params.WINDOWS_2019_DCAP_ICX_LABEL),
                     string(name: 'WINDOWS_NONSGX_CUSTOM_LABEL', value: params.WINDOWS_NONSGX_CUSTOM_LABEL),
+                    string(name: 'WS2022_DCAP_CFL_LABEL', value: params.WINDOWS_2022_DCAP_LABEL),
+                    string(name: 'WS2022_DCAP_ICX_LABEL', value: params.WINDOWS_2022_DCAP_ICX_LABEL),
+                    string(name: 'WINDOWS_2022_NONSGX_CUSTOM_LABEL', value: params.WINDOWS_2022_NONSGX_CUSTOM_LABEL),
```
Comment (CyanDevs) [inline .jenkins/infrastructure/e2e_testing.Jenkinsfile:110]: Please make this change in Jenkins CasC too

File: .jenkins/library/vars/globalvars.groovy [hunk 2]
```diff
@@ -16,16 +16,20 @@ import groovy.transform.Field
     "acc-v3-ubuntu-20.04":         env.UBUNTU_2004_ICX_CUSTOM_LABEL ?: "ACC-v3-2004",
     "acc-win2019-dcap":            env.WINDOWS_2019_DCAP_CUSTOM_LABEL ?: "SGXFLC-Windows-2019-DCAP",
     "acc-v3-win2019-dcap":         env.WINDOWS_2019_DCAP_ICX_CUSTOM_LABEL ?: "ACC-v3-SGXFLC-Windows-2019-DCAP",
+    "acc-win2022-dcap":            env.WINDOWS_2022_DCAP_CUSTOM_LABEL ?: "SGXFLC-Windows-2022-DCAP",
+    "acc-v3-win2022-dcap":         env.WINDOWS_2022_DCAP_ICX_CUSTOM_LABEL ?: "ACC-v3-SGXFLC-Windows-2022-DCAP",
     // Non SGX VMs
     "ubuntu-nonsgx":               env.UBUNTU_NONSGX_CUSTOM_LABEL ?: "nonSGX-ubuntu-2004",
     "ubuntu-nonsgx-20.04":         env.UBUNTU_NONSGX_CUSTOM_LABEL ?: "nonSGX-ubuntu-2004",
     "windows-nonsgx":              env.WINDOWS_NONSGX_CUSTOM_LABEL ?: "nonSGX-Windows",
+    "windows-2022-nonsgx":          env.WINDOWS_2022_NONSGX_CUSTOM_LABEL ?: "nonSGX-Windows-2022",
```
Comment (CyanDevs) [inline .jenkins/library/vars/globalvars.groovy:25]: nit: spacing

File: .jenkins/infrastructure/e2e_testing.Jenkinsfile (hunk 1 above)
Comment (salsal97) [reply .jenkins/infrastructure/e2e_testing.Jenkinsfile:110]: Added

File: .jenkins/infrastructure/e2e_testing.Jenkinsfile [hunk 3]
```diff
@@ -99,9 +102,12 @@ pipeline {
                     string(name: 'UBUNTU_2004_CFL_CUSTOM_LABEL', value: params.UBUNTU_2004_CFL_LABEL),
                     string(name: 'UBUNTU_2004_ICX_CUSTOM_LABEL', value: params.UBUNTU_2004_ICX_LABEL),
                     string(name: 'UBUNTU_2004_NONSGX_LABEL', value: params.UBUNTU_NONSGX_LABEL),
-                    string(name: 'WS2019_DCAP_CFL_LABEL', value: params.WINDOWS_2019_DCAP_LABEL),
```
Comment (CyanDevs) [inline .jenkins/infrastructure/e2e_testing.Jenkinsfile:102]: Please revert the name (not the values) changes back to what you had before as those were correct.

The names specifies the names of the parameters used in the openenclave-testing pipeline. The values are the parameters specified in this pipeline (i.e. lines 27-46 in this file).

What this section does is pass specific parameters from this pipeline into OpenEnclave-Testing pipeline to run, which is why the names are similar but different. The inconsistency can be fixed by changing all occurrences to be prefixed with either `WS*` or `WINDOWS_*` or something else, but that is outside the scope of this PR.

**Review by CyanDevs — APPROVED** [review]
LGTM. Great work! Please squash & sign-off as well, thanks

**Comment by CyanDevs** [comment]
Note that we should merge this only after we complete the infrastructure config and apply it.

File: .jenkins/pipelines/Azure/Windows/Jenkinsfile [hunk 4]
```diff
@@ -11,7 +11,8 @@ def testing_stages = [
...
     "XC Win2019 v2 clang-11 RelWithDebInfo ControlFlow snmalloc":            { tests.windowsCrossCompile(params.WS2019_DCAP_CFL_LABEL, 'clang-11', 'RelWithDebInfo', 'ControlFlow', '0', 'ON', 'ON') },
     "XC Win2019 v2 clang-11 Debug ControlFlow":                              { tests.windowsCrossCompile(params.WS2019_DCAP_CFL_LABEL, 'clang-11', 'Debug',          'ControlFlow', '0', 'ON', 'OFF') },
-    "XC Win2019 v3 clang-11 RelWithDebInfo ControlFlow":                     { tests.windowsCrossCompile(params.WS2019_DCAP_ICX_LABEL, 'clang-11', 'RelWithDebInfo', 'ControlFlow', '0', 'ON', 'OFF') }
+    "XC Win2019 v3 clang-11 RelWithDebInfo ControlFlow":                     { tests.windowsCrossCompile(params.WS2019_DCAP_ICX_LABEL, 'clang-11', 'RelWithDebInfo', 'ControlFlow', '0', 'ON', 'OFF') },
+    "XC Win2022 v2 clang-11 Debug ControlFlow":                              { tests.windowsCrossCompile(params.WS2022_DCAP_CFL_LABEL, 'clang-11', 'Debug',          'ControlFlow', '0', 'ON', 'OFF') }
```
Comment (radhikaj) [inline .jenkins/pipelines/Azure/Windows/Jenkinsfile:15]: Make this RelWithDebInfo

File: .jenkins/Jenkinsfile [hunk 5]
```diff
@@ -129,7 +129,7 @@ pipeline {
                                 string(name: 'DOCKER_TAG', value: params.DOCKER_TAG),
                                 string(name: 'UBUNTU_2004_CUSTOM_LABEL', value: globalvars.AGENTS_LABELS["acc-ubuntu-20.04"]),
                                 string(name: 'UBUNTU_NONSGX_CUSTOM_LABEL', value: globalvars.AGENTS_LABELS["ubuntu-nonsgx"]),
-                                string(name: 'WINDOWS_NONSGX_CUSTOM_LABEL', value: globalvars.AGENTS_LABELS["windows-nonsgx"]),
+                                string(name: 'WS2019_NONSGX_CUSTOM_LABEL', value: globalvars.AGENTS_LABELS["ws2019-nonsgx"]),
                                 string(name: 'OECI_LIB_VERSION', value: GLOBAL_OECI_LIB_VERSION),
```
Comment (radhikaj) [inline .jenkins/Jenkinsfile:133]: We should add a WS2022 non SGX stage as well. Can do in later PR

Comment (salsal97) [reply .jenkins/Jenkinsfile:133]: Sounds good

**Review by radhikaj — APPROVED** [review]

**Pull request merged**